### ISSUE TITLE
No way to override the global variables in the sub-charts out of the box.

### ISSUE THREAD

**Issue by albionb96** [helm/helm#13354]
During the implementation of an application I hit this issue, and I would like to know if there would be place for improvement in this direction.

Let's imagine a use-cases where you have one parent chart and one common variable which will be used from 3 of its sub-charts.

1) On one side I don't want to define the same variable with the same value three times in the parent chart values.yaml, 
2) on the other side I don't want to define the variable only under the "global" section in the parent chart because that would mean that my sub-charts are always depended on a parent chart and cannot be deployed on their own, becuase I need to always prefix that global variable with .Values.global.
3) There is no automatic variable precedence between the variables in the global section and the sub-chat variables. Global variables are prefixed with .Values.global and sub-chart values are prefixed only with .Values. You need an extra glue logic for that.

What do you think of this approach:

```
1) Define the common variable in the global section in the parent chart.

parentChartValues.yaml:

  global:
    myCommonVar: myCommonValue

2) Define the same variable in the sub-charts if you want to override the global, or in case you are going to deploy the sub-chart independently from parent. Otherwise leave it empty and global value should take precedence.

defaultSubchart1Values.yaml

  myCommonVar: mySpecialSubchart1Value


defaultSubchart2Values.yaml

  myCommonVar: mySpecialSubchart2Value


3) Create a common-library which will create a variable precedence between the variables under the global section and the ones in values.yaml of the sub-charts.
common-library-chart:

_common.tpl:

  {{- define "common.myCommonVar" -}}
    {{- default .Values.myCommonVar .Values.global.myCommonVar }}
  {{- end }}



4) Include the common library chart only in the parent helm chart/ or in all sub-chart independently.


5) Usage in the charts:

 {{- include "common.myCommonVar" -}} or {{- template"common.myCommonVar" -}}
```

**Comment by robertsirc** [comment]
It is an interesting proposal here. I like the premise of it. Have you suggested this for Helm 4?

**Comment by albionb96** [comment]
@robertsirc No I didn't propose anything, because my approach is also now possible to be implemented but you need to add the glue logic in a go template syntax as described above and that boilerplate is not very practical. My issue aimed to question if there is a better and more helm-native approach than this for the moment.

I later found many other tickets from the past requesting something like this but nothing happened in that direction until now: https://github.com/helm/helm/issues/12323#issuecomment-1774091608

**Comment by JasonShuyinta** [comment]
I stumbled upon the same situation, and one "dirty" way in which I resolved it was:
`
parentChartValues.yaml:

  global:
    myCommonVar: myCommonValue
---
defaultSubchart1Values.yaml

  global:
    myCommonVar: mySpecialSubchart1Value

---
defaultSubchart2Values.yaml

   global:
      myCommonVar: mySpecialSubchart2Value
`

This way you can deploy the subcharts independently, because the global.myCommonVar it's actually there. 
Now when I deploy the parent chart with its subcharts it is going to override the values in the subchart with the one in the parent chart. 

Basically we are "tricking" the subchart by defining a fake global variable when it's deployed independently, but overriding it when it's deployed as a subchart. 

It may not be clean but it does the trick
